modules/_node-scripts/check/tsc/index.mjs

```diff
@@ -0,0 +1,71 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+import path from 'path';
+import resolve from 'resolve';
+import os from 'os';
+
+import {getProjectDirs, getRootDir, SRC_TSCONFIG_PATH} from '../../util/constants.mjs';
+import fileExists from '../../util/fileExists.mjs';
+import forkModule from '../../util/forkModule.mjs';
+
+export default async function main() {
+	const cwd = path.resolve('.');
+	const rootDir = await getRootDir();
+
+	if (cwd === rootDir) {
+		const projectGroups = [[]];
+
+		const cpuCount = os.cpus().length;
+
+		for (const projectDir of await getProjectDirs()) {
+			if (!await fileExists(path.join(projectDir, SRC_TSCONFIG_PATH ))) {
+				continue;
+			}
+
+			let group = projectGroups[projectGroups.length-1];
+
+			if (group.length === cpuCount) {
+				group = [];
+
+				projectGroups.push(group);
+			}
+
+			group.push(projectDir);
+		}
+
+		console.log(`ℹ️ ${cpuCount} CPUs detected: launching tsc in groups of ${cpuCount} projects`);
+
+		for (const projectGroup of projectGroups) {
+			await Promise.all(
+				projectGroup.map(projectDir => {
+					console.log(`🕵️ Checking ${path.relative(rootDir, projectDir)}`);
+
+					return runTsc(projectDir);
```

Comment:
Is there no need to calculate the "ordering" or projects anymore? Before we had it run "parent" packages before running it's dependents.

Reply:
Since we are using references, `tsc` knows how to do it bc it rebuilds any dependency if necessary before checking a project.

I wonder how it manages the build directory from the POV of locking since we are running several `tsc`s in parallel writing to the same output, but I haven't seen any problems arising from that, so I assume it handles it correctly (usually other compilers use locking mechanisms in the output directories, so it's a reasonable assumption).